bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/DateTimeType.java

```diff
@@ -248,4 +250,85 @@ private ZonedDateTime parse(String value) throws DateTimeParseException {
 
         return date;
     }
+
+    public boolean isToday() {
+        return sameDay(ZonedDateTime.now());
+    }
+
+    public boolean isTomorrow() {
+        return sameDay(ZonedDateTime.now().plusDays(1));
+    }
+
+    public boolean isYesterday() {
+        return sameDay(ZonedDateTime.now().minusDays(1));
+    }
+
+    public boolean sameDay(DateTimeType other) {
+        return sameDay(other.zonedDateTime);
+    }
+
+    public boolean sameDay(ZonedDateTime other) {
+        return zonedDateTime.truncatedTo(ChronoUnit.DAYS)
+                .isEqual(other.withZoneSameInstant(zonedDateTime.getZone()).truncatedTo(ChronoUnit.DAYS));
+    }
+
+    public DateTimeType toToday() {
+        return shiftDaysFromToday(0);
+    }
+
+    public DateTimeType toTomorrow() {
+        return shiftDaysFromToday(1);
+    }
+
+    public DateTimeType toYesterday() {
+        return shiftDaysFromToday(-1);
+    }
+
+    private DateTimeType shiftDaysFromToday(int days) {
+        ZonedDateTime now = ZonedDateTime.now().plusDays(days);
+        return new DateTimeType(zonedDateTime.withYear(now.getYear()).withMonth(now.getMonthValue())
+                .withDayOfMonth(now.getDayOfMonth()));
+    }
+
+    public boolean isBefore(DateTimeType other) {
+        return zonedDateTime.isBefore(other.zonedDateTime);
+    }
+
+    public boolean isAfter(DateTimeType other) {
+        return zonedDateTime.isAfter(other.zonedDateTime);
+    }
+
```

Comment:
We might also need the versions of isBefore and isAfter that accept a ZonedDateTime

Reply:
These already [exists](https://docs.oracle.com/javase/8/docs/api/java/time/chrono/ChronoZonedDateTime.html#isBefore-java.time.chrono.ChronoZonedDateTime-), reason why I thought it wasn't needed.

Reply:
Basically it's for convenience, so you can do DateTimeItem.state.isBefore(zdt) instead of DateTimeItem.state.getZonedDateTime().isBefore(zdt), and also for uniformity since you are adding the ability to do 
DateTimeItem.state.isBefore(anotherDateTimeItem), but, I'll leave the decision whether to have it to the maintainers.

Reply:
Added